File: Front End/src/app/modules/donations/models/donation.ts
import { Currency } from "./currency";
import { PoliticalEntity } from "./political-entity";

export class Donation {
    // id: String;
    id: Number;
    name: string;
    amount: Number;
    entityId: Number;
    destination: string;
    condition?: string;
    currencyId: Number;
    exchangeRate: Number;
    
    constructor() {
        this.id = 0;
    }
}

File: Back End/Donations.Entities/Models/Donation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Donations.Entities.Models
{
    public class Donation
    {
        [Key]
        public int Id { get; set; } // Primary Key

        public string Name { get; set; }

        public decimal Amount { get; set; }

        public int EntityId { get; set; } // Foreign Key to the Entity model

        public string Destination { get; set; }

        public string Condition { get; set; }

        public int CurrencyId { get; set; } // Foreign Key to the Currency model

        public decimal ExchangeRate { get; set; }

        // Navigation Properties (for Entity Framework relationships)
        //public Entity Entity { get; set; }

        //public Currency Currency { get; set; }
    }
}
